src/service/db/session.rs

```diff
@@ -13,119 +13,164 @@
-    service::db::{self},
+use infra::{
+    db::ORM_CLIENT,
+    table::entity::{prelude::Sessions, sessions},
 };
+use sea_orm::{ActiveModelTrait, ColumnTrait, EntityTrait, QueryFilter, Set};
 
-// DBKey to set settings for an org
-pub const USER_SESSION_KEY: &str = "/user_sessions/";
+use crate::common::infra::config::USER_SESSIONS;
 
 pub async fn get(session_id: &str) -> Result<String, anyhow::Error> {
+    // Check cache first
     if let Some(val) = USER_SESSIONS.get(session_id) {
         return Ok(val.to_string());
     }
 
-    // get from db
+    // Cache miss - read from database
     log::warn!("Cache miss for user session, read from db: {}", session_id);
-    let val = db::get(&format!("{USER_SESSION_KEY}{session_id}")).await?;
-    // json format: convert bytes to string and trim any quotes
-    let val = String::from_utf8(val.to_vec())
-        .unwrap()
-        .trim_matches('"')
-        .to_string();
-    // cache it in memory
-    if !val.is_empty() {
-        USER_SESSIONS.insert(session_id.to_string(), val.to_string());
+    let db = ORM_CLIENT
+        .get()
+        .ok_or_else(|| anyhow::anyhow!("ORM client not initialized"))?;
+
+    let session = Sessions::find()
+        .filter(sessions::Column::SessionId.eq(session_id))
+        .one(db)
+        .await?;
+
+    match session {
+        Some(session) => {
+            let access_token = session.access_token.clone();
+            // Cache it in memory
+            if !access_token.is_empty() {
+                USER_SESSIONS.insert(session_id.to_string(), access_token.clone());
+            }
+            Ok(access_token)
+        }
+        None => Err(anyhow::anyhow!("Session not found: {}", session_id)),
     }
-    Ok(val)
 }
 
 pub async fn set(session_id: &str, val: &str) -> Result<(), anyhow::Error> {
-    db::put(
-        &format!("{USER_SESSION_KEY}{session_id}"),
-        json::to_vec(&val).unwrap().into(),
-        db::NEED_WATCH,
-        None,
-    )
-    .await?;
+    let db = ORM_CLIENT
+        .get()
+        .ok_or_else(|| anyhow::anyhow!("ORM client not initialized"))?;
+
+    let now = chrono::Utc::now().timestamp_micros();
+
+    // Check if session exists
+    let existing = Sessions::find()
+        .filter(sessions::Column::SessionId.eq(session_id))
+        .one(db)
+        .await?;
+
+    if let Some(existing) = existing {
+        // Update existing session
+        let mut active_model: sessions::ActiveModel = existing.into();
+        active_model.access_token = Set(val.to_string());
+        active_model.updated_at = Set(now);
+        active_model.update(db).await?;
+    } else {
+        // Insert new session
+        let active_model = sessions::ActiveModel {
+            id: sea_orm::ActiveValue::NotSet,
+            session_id: Set(session_id.to_string()),
+            access_token: Set(val.to_string()),
+            created_at: Set(now),
+            updated_at: Set(now),
+        };
+        active_model.insert(db).await?;
```

Comment:
We generally write the all the direct code call to db functions inside the `infra/src` in its own file, and then use those functions here. Can you do the same for the sessions as well. The same code, just create functions inside the infra/src and use them here.